src/3d/qgsshadowrenderingframegraph.cpp

```diff
@@ -0,0 +1,342 @@
+/***************************************************************************
+  qgsshadowrenderingframegraph.cpp
+  --------------------------------------
+  Date                 : August 2020
+  Copyright            : (C) 2020 by Belgacem Nedjima
+  Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsshadowrenderingframegraph.h"
+
+#include "qgsdirectionallightsettings.h"
+#include "qgscameracontroller.h"
+#include "qgsrectangle.h"
+#include "qgspostprocessingentity.h"
+#include "qgspreviewquad.h"
+
+Qt3DRender::QFrameGraphNode *QgsShadowRenderingFrameGraph::constructTexturesPreviewPass()
+{
+  mPreviewLayerFilter = new Qt3DRender::QLayerFilter;
+  mPreviewLayerFilter->addLayer( mPreviewLayer );
+
+  mPreviewRenderStateSet = new Qt3DRender::QRenderStateSet( mPreviewLayerFilter );
+  mPreviewDepthTest = new Qt3DRender::QDepthTest;
+  mPreviewDepthTest->setDepthFunction( Qt3DRender::QDepthTest::Always );
+  mPreviewRenderStateSet->addRenderState( mPreviewDepthTest );
+  mPreviewCullFace = new Qt3DRender::QCullFace;
+  mPreviewCullFace->setMode( Qt3DRender::QCullFace::NoCulling );
+  mPreviewRenderStateSet->addRenderState( mPreviewCullFace );
+
+  return mPreviewLayerFilter;
+}
+
+Qt3DRender::QFrameGraphNode *QgsShadowRenderingFrameGraph::constructForwardRenderPass()
+{
+  mForwardRenderLayerFilter = new Qt3DRender::QLayerFilter( mMainCameraSelector );
+  mForwardRenderLayerFilter->addLayer( mForwardRenderLayer );
+
+  // TODO: make the width and height change dynamically as the 3D viewer is resized
+  int width = 1024;
+  int height = 768;
```

Comment:
I think before there wasn't a way to resize the resolution when the 3D viewer is resized because there was no access for it in the QForwardRenderer, this can be made into signals and slots to resize the textures accordingly.
Please tell me if I misunderstood something about the QForwardRenderer and the resolution.